refactor(docs-infra): remove obsolete ViewEngine code path from `ElementsLoader`

gkalpak opened the pull request:
Remove a code path that was only reached in ViewEngine mode and is now (with the removal of ViewEngine) obsolete.

mary-poppins commented:
You can preview 2ec3ef8 at https://pr44293-2ec3ef8.ngbuilds.io/.

AndrewKushnir commented on aio/src/app/custom-elements/elements-loader.ts:
```diff
@@ -52,21 +45,8 @@ export class ElementsLoader {
+        (modulePathLoader() as Promise<Type<WithCustomElementComponent>>)
+          .then(elementModule => this.compiler.compileModuleAsync(elementModule))
           .then(elementModuleFactory => {
             const elementModuleRef = elementModuleFactory.create(this.moduleRef.injector);
             const injector = elementModuleRef.injector;
```
Just a drive by comment: I think you could get rid of the `Compiler` symbol here as well by leveraging the https://angular.io/api/core/createNgModuleRef function (you may not need to resolve/compile the module factory).

gkalpak replied:
🎉

Thank you so much, @AndrewKushnir ❤️
I looked for something like this, but I couldn't find anything.

AndrewKushnir replied:
Wlcm :)

gkalpak committed:
refactor(docs-infra): remove obsolete ViewEngine code path from `ElementsLoader`

Remove a code path that was only reached in ViewEngine mode and is now
(with the removal of ViewEngine) obsolete.

gkalpak committed:
refactor(docs-infra): replace use of deprecated `Compiler` with `createNgModuleRef()`

Remove a use of the deprecated `Compiler` class in `ElementsLoader` and
use the `createNgModuleRef()` function instead.

mary-poppins commented:
You can preview e814bab at https://pr44293-e814bab.ngbuilds.io/.

mary-poppins commented:
You can preview be952ea at https://pr44293-be952ea.ngbuilds.io/.

dylhunn commented:
This PR was merged into the repository by commit f7ea5249deb98d691cac64edc4ae38404ee47a06.
